qiskit/transpiler/passes/optimization/optimize_1q_gates.py

```diff
@@ -206,6 +209,9 @@ def run(self, dag):
                 else:
                     raise TranspilerError('It was not possible to use the basis %s' % self.basis)
 
+            # safer to keep phase on op instead of dag (e.g. if removed)?
+            if new_op.definition is not None:
+                new_op.definition.global_phase = right_global_phase
```

Comment:
Will this drop the global phase if `new_op` is a `U3Gate`?

Reply:
Is it possible for `new_op` to have a global phase? In the declarations above it did not appear to.

Reply:
It's `definition` could. I'm wondering what happens to `right_global_phase` if `new_op.definition is None` (like for a `U3Gate`).